app/controllers/carts_controller.rb

```diff
@@ -13,10 +12,41 @@ def show
   end
 
   def destroy
-    item = Item.find(params[:item_id])
-    @cart.remove_item(item.id)
-    session[:cart] = @cart.contents
-    flash[:success] = "Succesfully removed #{view_context.link_to(item.title, item)} from your cart."
+    if params[:remove].nil?
+      @cart.decrease_item_quantity(item.id)
```

Comment:
So here I think we are missing a subtlety of one of the user stories:

If we are completely removing this item, we need to show the remove_message

It should be a quick fix: 
1. either remove_message, then decrement by one if qty is not zero.
2. check to see if qty is 1, then remove_message, then decrement